src/main/java/com/google/sps/data/Places.java

```diff
@@ -145,17 +149,63 @@
     return stream;
   }
 
+  /**
+   * This function getAllPlaces retrieved from a Places Search API using the API
+   * and the user's location.
+   * @return Set<PlacesSearchResult[]> This returns a set of arrays of Places Search Results.
+   */
+  public static Set<HiddenGem> getHiddenPlaces(Set<PlacesSearchResult> places) {
+    GeoApiContext context = new GeoApiContext.Builder()
+      .apiKey(GetConfigProperties.getApiKey())
+      .build();
+
+    return convertToHiddenGem(context, places);
+  }
+
+  /**
+   * This converts a set of places into a set of hidden gems. 
+   * Iterate over the set of places and find the additional details, then create
+   * the hidden gem object by populating all the relevant data in the constructor.
+   * @param places            The set of places search results to be converted.
```

Comment:
Update params

Reply:
I added in the param from the context? Is that what you meant when you commented update params? Just wanted to double check